### PR TITLE
Updating projects, dockerfile and ci to dotnet 6

### LINKED ISSUE #3: Upgrade to .NET 6
I'm having a bear of a time getting this to compile on my Mac since the projects target .NET 5.0 which is deprecated. Update all the csproj files to use .NET 6.0 (keep net standard and 3.1 as those are LTS I believe).

### PR DESCRIPTION
**Pull request by lambchop4prez**
Closes #3 

Still not quite done yet, but I wanted to  get the PR up to show that I'm working on it. It looks like CI is broken, so I'll fix that too when I get a chance.

### CODE REVIEW

File: src/HotPotato.AspNetCore.Middleware/HotPotato.AspNetCore.Middleware.csproj [hunk 1]
```diff
@@ -2,9 +2,9 @@
 
   <PropertyGroup>
     <OutputType>Library</OutputType>
-    <TargetFrameworks>netcoreapp3.1;net5.0;netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>net6.0</TargetFrameworks>
```
Comment (davidmbillie) [inline src/HotPotato.AspNetCore.Middleware/HotPotato.AspNetCore.Middleware.csproj:5]: Would it be possible to keep .NET Standard in here? If it's messing too much with the package versions too much, I'm not against removing it.

**Comment by davidmbillie** [comment]
I have a local branch for updating some things in our E2E tests, and lot of the package disparities were making things difficult. I'll wait to push that up when this is merged.

**Review by davidmbillie — APPROVED** [review]

File: src/HotPotato.AspNetCore.Middleware/HotPotato.AspNetCore.Middleware.csproj (hunk 1 above)
Comment (lambchop4prez) [reply src/HotPotato.AspNetCore.Middleware/HotPotato.AspNetCore.Middleware.csproj:5]: Yeah, I'm going to revisit this now that I've read this page: https://learn.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-7.0

Specifically:

> However, if you want to share code between .NET Framework and any other .NET implementation, such as .NET Core, your library should target .NET Standard 2.0. [No new versions of .NET Standard will be released](https://devblogs.microsoft.com/dotnet/the-future-of-net-standard/), but .NET 5, .NET 6, and all future versions will continue to support .NET Standard 2.1 and earlier.

So I think we may want to target netstandard2.1 at least for the main components other folks will be integrating with (Middleware, Core, and OpenAPI plus any supporting libraries). I think the host is fine to keep on net6, but I'm gonna play around with that a bit.

**Comment by lambchop4prez** [comment]
@davidmbillie I set Middleware, Core, OpenAPI, and NunitHelpers to netstandard 2.1. But I kept the Host and Test Extensions on net6 since those use libraries that don't support netstandard. For the tests, I kept them on net6 as well, since they don't matter as much. Pull these changes in and let me know if you are still having issues.

**Comment by davidmbillie** [comment]
> @davidmbillie I set Middleware, Core, OpenAPI, and NunitHelpers to netstandard 2.1. But I kept the Host and Test Extensions on net6 since those use libraries that don't support netstandard. For the tests, I kept them on net6 as well, since they don't matter as much. Pull these changes in and let me know if you are still having issues.

So here's the goofiness I'm seeing - we have that HotPotato.NetFramework.Test project written in .NET Framework 4.7.2 that's incompatible with .NET Standard 2.1. I'm not quite sure why it would be compatible with 2.0 but not 2.1 
image

**Comment by davidmbillie** [comment]
This looks good to go - everything's building and passing locally for me. I can create a separate PR for the minor E2E test edits I made once this is merged.

**Pull request merged**